torchtune/modules/position_embeddings.py

```diff
@@ -107,8 +109,9 @@ def forward(self, x: Tensor, input_pos: Optional[Tensor] = None) -> Tensor:
         xshaped = x.float().reshape(*x.shape[:-1], -1, 2)
 
         # reshape the cache for broadcasting
-        # tensor has shape [1, s, 1, n_d // 2, 2]
-        rope_cache = rope_cache.view(1, xshaped.size(1), 1, xshaped.size(3), 2)
+        # tensor has shape [b, s, 1, n_d // 2, 2] if packed samples,
```

Comment:
I dont think I understand this?

Reply:
previously the rope embeddings are the same for each sample within a batch because they will all go to max seq len, but now the position ids may be different for each sample because each sample has multiple sequences of different lengths, so now the rope cache has an added batch dimension.

Reply:
Yeah maybe I'm being a little dense here, but I don't understand how we index into `self.cache` with the 2D input_pos tensor and it just works out of the box.

Reply:
pytorch magic
<img width="607" alt="image" src="https://github.com/pytorch/torchtune/assets/33648637/0016d59b-f480-4211-8a07-1b9813cdcacb">